plopfile.js

```diff
@@ -0,0 +1,143 @@
+    prompts: [
+      {
+        type: "input",
+        name: "name",
+        message: "What is the name of your test app?",
+        validate: name => Boolean(name.trim()),
+        transformer: name => name.replace(/[^-\w]/g, "")
+      },
+      {
+        type: "list",
+        name: "platforms",
+        message: "Which platforms do you need test apps for?",
+        choices: ["All", "Android", "iOS", "macOS", "Windows"]
+      }
+    ],
+    actions: (/** @type {InputData} */ { name, platforms }) => {
+      const path = require("path");
+      const templateDir = path.dirname(
+        require.resolve("react-native/template/package.json")
+      );
+      const actions = [
+        {
+          type: "add",
+          path: ".watchmanconfig",
+          templateFile: path.join(templateDir, "_watchmanconfig")
+        },
+        {
+          type: "add",
+          path: "app.json",
+          template: JSON.stringify(
+            {
+              name,
+              displayName: name,
+              components: {
+                [name]: {
+                  displayName: name
+                }
+              },
+              resources: ["assets", "main.jsbundle"]
+            },
+            undefined,
+            2
+          )
+        },
+        {
+          type: "add",
+          path: "babel.config.js",
+          templateFile: require.resolve("react-native/template/babel.config.js")
+        },
+        {
+          type: "add",
+          path: "index.js",
+          templateFile: require.resolve("react-native/template/index.js")
+        },
+        {
+          type: "add",
+          path: "metro.config.js",
+          templateFile: require.resolve("react-native/template/metro.config.js")
+        },
+        {
+          type: "add",
+          path: "package.json",
+          templateFile: require.resolve("react-native/template/package.json"),
+          transform: template => {
+            const {
+              name: testAppPackageName,
+              version: testAppPackageVersion
+            } = require("./package.json");
+            const packageJson = JSON.parse(template);
+            const devDependencies = {
+              ...packageJson.devDependencies,
+              [testAppPackageName]: testAppPackageVersion
+            };
+            return JSON.stringify(
+              {
+                ...packageJson,
+                name,
+                devDependencies: Object.keys(devDependencies)
+                  .sort()
+                  .reduce(
+                    (deps, key) => ({ ...deps, [key]: devDependencies[key] }),
+                    {}
+                  )
+              },
+              undefined,
+              2
+            );
+          }
+        }
+      ];
+
+      const exclusive = platforms !== "All";
+      if (!exclusive || platforms === "iOS") {
+        actions.push({
+          type: "add",
+          path: exclusive ? "Podfile" : "ios/Podfile",
```

Comment:
This doesn't quite work unless we take a direct dependency on `@react-native-community/cli` 3.x, or simply bump React Native to 0.61. Still have to figure out whether we can simply do the former.

Reply:
Took a dependency on `@react-native-community/cli` 4.x for now. It'll have to be removed when we bump React Native.